### PR TITLE
Rfink/jdk7 skim

### PR DESCRIPTION
**Pull request by uschi2000**
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/300)

### CODE REVIEW

**Comment by uschi2000** [comment]
Ping @markelliot

File: extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/PathDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
```
Comment (markelliot) [inline extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/PathDeserializer.java:2]: this should be the Apache 2 header

File: extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/ShimJdk7Module.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
```
Comment (markelliot) [inline extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/ShimJdk7Module.java:2]: same here

File: extras/jackson-support/src/test/java/com/palantir/remoting1/ext/jackson/PathDeserializderTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.Test;
+
+public final class PathDeserializderTest {
```
Comment (markelliot) [inline extras/jackson-support/src/test/java/com/palantir/remoting1/ext/jackson/PathDeserializderTest.java]: possibly a copy of my typo, but probably should be PathSerDeTest

Comment (markelliot) [reply extras/jackson-support/src/test/java/com/palantir/remoting1/ext/jackson/PathDeserializderTest.java]: or Jdk7SerDeTest

**Comment by markelliot** [comment]
g2g except for nits

**Comment by uschi2000** [comment]
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions.

---

_[extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/PathDeserializer.java, line 2 at r1](https://reviewable.io:443/reviews/palantir/http-remoting/300#-KWer72uV3K7VOjqo2KH-r1-2:-KWk6ggLNA4yyq82vQrB:b-xt727g) ([raw file](https://github.com/palantir/http-remoting/blob/14458ca5dec6d34e19daedad7e66e857225ad30b/extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/PathDeserializer.java#L2)):_
<details><summary>

<i>Previously, markelliot (Mark Elliot) wrote…</i></summary>



> this should be the Apache 2 header
> </details>
> Copied this from your PR. What header exactly do you mean? The exact one from FasterXML (i.e., without mentioning Palantir)?

---

_[extras/jackson-support/src/test/java/com/palantir/remoting1/ext/jackson/PathDeserializderTest.java, line 27 at r1](https://reviewable.io:443/reviews/palantir/http-remoting/300#-KWer72uV3K7VOjqo2KK-r1-27:-KWk6tq4tH6xld8T3BSi:b-72l0zz) ([raw file](https://github.com/palantir/http-remoting/blob/14458ca5dec6d34e19daedad7e66e857225ad30b/extras/jackson-support/src/test/java/com/palantir/remoting1/ext/jackson/PathDeserializderTest.java#L27)):_
<details><summary>

<i>Previously, markelliot (Mark Elliot) wrote…</i></summary>



> or Jdk7SerDeTest
> </details>
> my bug. fixed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/300)_

**Comment by markelliot** [comment]
the header should include the Apache license beneath our copyright, not just as part of the other block.

**Comment by uschi2000** [comment]
done

---

Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/300#-:-KWodn7Z409HQZ9pi63k:bb7f2mf)_

**Comment by uschi2000** [comment]
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

_[extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/ShimJdk7Module.java, line 2 at r1](https://reviewable.io:443/reviews/palantir/http-remoting/300#-KWer72uV3K7VOjqo2KI-r1-2:-KWodpPYI49jxNT_v0bT:b-896fix) ([raw file](https://github.com/palantir/http-remoting/blob/14458ca5dec6d34e19daedad7e66e857225ad30b/extras/jackson-support/src/main/java/com/palantir/remoting1/ext/jackson/ShimJdk7Module.java#L2)):_
<details><summary>

<i>Previously, markelliot (Mark Elliot) wrote…</i></summary>



> same here
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/300)_

**Comment by uschi2000** [comment]
rebased.

---

Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/300#-:-KWoeCs7NKW5GO97ZUih:bmaduqb)_

**Comment by markelliot** [comment]
:lgtm:

---

Reviewed 16 of 20 files at r1, 2 of 2 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/300#-:-KWoehPQ6vCTKJnUGUCF:bnfp4nl)_

**Pull request merged**